packages/data/lib/service/api_service.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:dio/dio.dart';
+
+abstract class ApiService {
+  factory ApiService(Dio dio) => _ApiService(dio);
+
+  Future<Response<T>> get<T>({
+    required String path,
+    Map<String, dynamic>? queryParameters,
+    Options? options,
+    ProgressCallback? onReceiveProgress,
+  });
+
+  setInterceptor({required Interceptor interceptor});
```

Comment:
FYI - usually the difference between add and set methods is that when we call both methods twice, the set method will reset the previous value (will be saved last value) and add method will add new value (will be saved previous and last values). so, in this case it's better to call method addInterceptor

Reply:
corrected